packages/app/src/server/service/page.ts

```diff
@@ -562,8 +562,8 @@ class PageService {
     await this.updateDescendantCountOfAncestors(renamedPage._id, nToIncrease, false);
 
     // Remove leaf empty pages if not moving to under the ex-target position
-    if (!this.isRenamingToUnderTarget(page.path, newPagePath)) {
-    // remove empty pages at leaf position
+    if (page.path !== newPagePath && !this.isRenamingToUnderTarget(page.path, newPagePath)) {
```

Comment:
Can you briefly explain?

Reply:
I added this because I noticed that page.path always equals newPagePath when moving pages so I skip executing `removeLeafEmptyPagesRecursively` if they are the same

On CI I've also seen tests that require removing blank pages that are in the leaf below the target page. Especially for recursively generated pages like `pages/a/b/c`. If the pages are created one by one in a hierarchy, the problem does not occur
Please give your suggestions